141 add dto response to user

karolina-svensson opened the pull request:
Link to documentation: https://documenter.getpostman.com/view/40894421/2sAYXFjdjp

## Notes & Questions For The Reviewer
Added proper responses to User getAll, getById and getByEmail as we were recieving the entire User object before. Also modified structure in methods getUserContacts and activateDeactivateUser to work after I implemented method for getById.


## What has been changed?
Added response to:
Get all users
Get user by ID
Get user by e-mail
Also check responses for Get User Contacts and Activate Deactivate User as I mad minor changes to these methods.

GET    /api/users/{id}     - Get user by id
GET    /api/users          - List all users
GET    /api/users/email/{email} - Get user by e-mail
GET    /api/users/contact/{contact} - Get user contacts
PATCH /api/users/{id}    - Activate/Deactivate user

## Why did I make these changes?
We needed better responses for some user endpoints.

## How can others test my code?
check out branch
start server
test every endpoint in postman
Test negativ scenarios, for example making error or trying to find resources that dont exist.

Checklist before sending in you PR
Check each box by adding an X to [ ] - [x]
[x] - I have tested my code and it works the way i want it to
[x] - My code follows our code standards (tex validation, names, etc)
[x] - I have commented komplicated parts of code (if needed)
[x] - I have uppdated dokumentation (if needed)

Remember that:
Be humble and open for feedback
Answer the comments quickly
Its okay to make mistakes

karolina-svensson committed:
created user response

karolina-svensson committed:
added UserResponse method to getAllUsers

karolina-svensson committed:
changed method for getting user by email

karolina-svensson committed:
modified getUserContacts in UserService

karolina-svensson committed:
modified activateDeactivateUser in UserService

Ollie1994 reviewed (changes requested):
![image](https://github.com/user-attachments/assets/0af9923c-24f6-42e3-a266-e265892eda8b)
![image](https://github.com/user-attachments/assets/f312d54b-3abe-4390-9f5c-1daf83ff9cac)
![image](https://github.com/user-attachments/assets/f302b5b1-f614-4943-8705-13174d5d5e6f)

karolina-svensson committed:
made getUserByEmail return one email instead of list

Ollie1994 commented:
![image](https://github.com/user-attachments/assets/abea682f-e9a4-4ffc-9933-319332b81697)

![image](https://github.com/user-attachments/assets/7b432551-5fb6-4098-9339-ec77624120ce)

karolina-svensson committed:
added username in convertToGetSingleUserResponse

karolina-svensson committed:
fixed conflicts

karolina-svensson commented:
added username in convertToGetSingleUserResponse (and String city in GetSingleUserResponse constructor that I had forgotten as well...). Also fixed conflicts. Here is what my postman response was after my changes. I realize the Ollie94 user has rating as null because that user does not have a rating object in our database, so I included a user that did in my picture as well.
![Skärmbild 2025-03-12 101648](https://github.com/user-attachments/assets/e95ef6f9-2137-4155-a9cc-cbf4ef0511a1)
![Skärmbild 2025-03-12 101827](https://github.com/user-attachments/assets/623272c8-c83c-4a6e-9c75-0d91e010f12c)

Ollie1994 reviewed (approved):
Funkar !
Bra jobbat!
